Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -242,9 +243,9 @@ private class _BSONReferencingEncoder: _BSONEncoder {
 
     /// Finalizes `self` by writing the contents of our storage to the referenced encoder's storage.
     deinit {
-        let value: BSONValue?
+        let value: BSONValue
         switch self.storage.count {
-        case 0: value = nil
+        case 0: value = NSNull()
```

Comment:
it appears that this case returns an empty `NSDictionary` in `JSONEncoder`. should this be an empty document? 

I realize that I am the one who put nil here initially but that may have been wrong. I don't think I ever wrote any tests for _BSONReferencingEncoder or Decoder... which was a poor choice.... just opened [SWIFT-263](https://jira.mongodb.org/browse/SWIFT-263) about that